src/main/java/org/broadinstitute/hellbender/utils/nio/NioFileCopier.java

```diff
@@ -0,0 +1,624 @@
+        final double   estTimeRemaining_ms      = remainingFileSize_bytes / bytesPerMillisecond;
+        final Duration estTimeRemainingDuration = Duration.ofMillis((long) estTimeRemaining_ms);
+
+        logger.info(
+                String.format("    Transfer: % 2.2f%% complete.  Est. time remaining: %s (@%3.02f kbps)",
+                        progressValue,
+                        formatMillisecondsTime(estTimeRemainingDuration.toMillis()),
+                        bytesPerMillisecond / KB_TO_BYTES * MS_TO_SEC
+                )
+        );
+    }
+
+    private void logProgressVerbose(final double progressValue, final long totalBytesRead, final double bytesPerMillisecond) {
+
+        // Get the remaining time estimate:
+        final long     remainingFileSize_bytes  = srcFileSize - totalBytesRead;
+        final double   estTimeRemaining_ms      = remainingFileSize_bytes / bytesPerMillisecond;
+        final Duration estTimeRemainingDuration = Duration.ofMillis((long) estTimeRemaining_ms);
+
+        final Instant endTime = Instant.now().plus(estTimeRemainingDuration);
+
+        logger.info(
+                String.format("    Transfer: % 2.2f%% complete (%" + srcFileSizeNumDigits + "d bytes; %6s).  Est. time remaining: %s (Complete time: %s) (@%3.02f kbps)",
+                        progressValue,
+                        totalBytesRead,
+                        FileUtils.byteCountToDisplaySize(totalBytesRead),
+                        formatMillisecondsTime(estTimeRemainingDuration.toMillis()),
+                        endTime.atZone(ZoneId.systemDefault()).toLocalDateTime().toString(),
+                        bytesPerMillisecond / KB_TO_BYTES * MS_TO_SEC
+                )
+        );
+    }
+
+    private void initializeCopyProgress(final long startTime_ns) {
+
+        // Track the time it takes to download each chunk:
+        lastProgressTime_ns = startTime_ns;
+
+        // Keep track of the total number of bytes read and our progress value:
+        totalBytesRead = 0;
+        progressBytesRead = 0;
+        lastProgressValue = 0;
+    }
+
+    private void updateCopyProgress(final int bytesRead) {
+
+        // Only bother with all this if we're logging in the first place:
+        if ( (!isSilentCopy()) && isDoLogProgress() ) {
+
+            // Update our progress counters:
+            totalBytesRead += bytesRead;
+            progressBytesRead += bytesRead;
+
+            // Get our progress percentage:
+            final double rawProgressValuePercent = ((double) totalBytesRead / (double) srcFileSize) * 100.0;
+
+            // Round our progress to nearest PROGRESS_DISPLAY_PERCENT_INCREMENT:
+            final double progressValue = PROGRESS_DISPLAY_PERCENT_INCREMENT * (Math.floor(Math.abs(rawProgressValuePercent / PROGRESS_DISPLAY_PERCENT_INCREMENT)));
+
+            // Output our progress if we're ready for it:
+            if ( progressValue != lastProgressValue ) {
```

Comment:
It seems like this might either update to fast on a fast download or too little on a slow one.  You might want to do something like the progress meter in gatktool does and only output every N seconds.  (and ultimately we want to make the progress meters async so they update when nothing is happening as well...).

Reply:
That's true.  I think I'd like to wait to refactor to timed output until it spins off threads to copy and to log progress.

As an approximation, I'll make it choose a delta % value for which to log based on the file size.